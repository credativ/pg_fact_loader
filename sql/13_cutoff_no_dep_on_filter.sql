SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';

/***
Based on our config, no actual changes will be processed based on these updates.
But we still want the queue to be cleared.
 */
UPDATE test.customers SET customer_number = customer_number||'1';

SELECT COUNT(1) FROM test_audit_raw.customers_audit;

SELECT test.tick();
SELECT fact_loader.worker() FROM generate_series(1,6);

--Should now handle dep fact tables
SELECT test.tick();
SELECT fact_loader.worker() FROM generate_series(1,6);
SELECT test.tick();
SELECT fact_loader.worker() FROM generate_series(1,6);

SELECT fact_loader.purge_queues('0 seconds'::INTERVAL);
SELECT COUNT(1) FROM test_audit_raw.customers_audit;

