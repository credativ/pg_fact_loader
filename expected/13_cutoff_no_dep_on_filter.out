SET client_min_messages TO warning;
--This is for testing functionality of timezone-specific timestamps
SET TIMEZONE TO 'America/Chicago';
/***
Based on our config, no actual changes will be processed based on these updates.
But we still want the queue to be cleared.
 */
UPDATE test.customers SET customer_number = customer_number||'1';
SELECT COUNT(1) FROM test_audit_raw.customers_audit;
 count 
-------
     9
(1 row)

SELECT test.tick();
 tick 
------
 
(1 row)

SELECT fact_loader.worker() FROM generate_series(1,6);
 worker 
--------
 t
 t
 t
 t
 t
 t
(6 rows)

--Should now handle dep fact tables
SELECT test.tick();
 tick 
------
 
(1 row)

SELECT fact_loader.worker() FROM generate_series(1,6);
 worker 
--------
 t
 t
 t
 t
 t
 t
(6 rows)

SELECT test.tick();
 tick 
------
 
(1 row)

SELECT fact_loader.worker() FROM generate_series(1,6);
 worker 
--------
 t
 t
 t
 t
 t
 t
(6 rows)

SELECT fact_loader.purge_queues('0 seconds'::INTERVAL);
 purge_queues 
--------------
 
(1 row)

SELECT COUNT(1) FROM test_audit_raw.customers_audit;
 count 
-------
     0
(1 row)

